superset/connectors/druid/models.py

```diff
@@ -1134,7 +1137,11 @@ def run_query(  # noqa / druid
             pre_qry = deepcopy(qry)
             if timeseries_limit_metric:
                 order_by = timeseries_limit_metric
-                pre_qry['aggregations'] = self.get_aggregations([timeseries_limit_metric])
+                aggs_dict, post_aggs_dict = DruidDatasource.aggs_and_post_aggs(
+                    [timeseries_limit_metric],
+                    metrics_dict)
+                pre_qry['aggregations'].update(aggs_dict)
```

Comment:
I'm unclear here as to why `.update` as opposed to direct affectations for both 1143 and 1144

Reply:
For two-phase queries, replacement is fine since the `pre_qry`s only purpose is to query for the top N groups. For single-phase queries, the `pre_qry` is the only query that runs, so the existing metrics that are queried for must be preserved as well.